doc/Programming-guide/Your-first-Kaa-application/index.md

```diff
@@ -767,43 +830,53 @@ Sampled temperature: 31
 Sampled temperature: 28
 Sampled temperature: 28
 ```
-Note that temperature value must be sampled once per second as stated in the configuration scheme.
-Refer to the [Troubleshooting](#troubleshooting) section if something goes wrong.
+The temperature value is sampled once per second as specified in the configuration schema.
+If you do not get the expected output, see [Debugging and troubleshooting]({{root_url}}Programming-guide/Debugging-and-troubleshooting).
 
-## Retrieving collected data
+### Retrieve collected data
 
-To obtain temperature data stored on the server, following steps should be performed.
+To obtain the temperature data stored on the server:
 
-1. Grab application token. It is a token that you can copy from the main window
-    of the application the Administration UI.
+1. Copy the application token from your new application page.
 
     ![Generate SDK](attach/app_token.png)
-1. Login to your machine running Kaa server (in case you are using sandbox,
-    default username and password are `kaa`/`kaa`).
-1. Start MongoDB shell
+
+2. Log in to your machine running Kaa server or to your Sandbox.
+For the Sandbox, use the default **kaa** username and **kaa123** password.
```

Comment:
The way how user should log in to the Sandbox is not clear here. User should login to the Sandbox VM using ssh with kaa/kaa credentials.